mitodl/ocw-studio
# Cleaning up collaborator api

#### Pre-Flight checklist

#### What are the relevant tickets?
Fixes #81 

#### What's this PR do?
- Moves data mutations into `WebsiteCollaboratorSerializer`
- Simplified `WebsiteCollaboratorViewSet.get_queryset` down to a single query definition
- Removed `update` / `create` methods on `WebsiteCollaboratorViewSet` which were now unnecessary
- Simplified validation in `WebsiteCollaboratorSerializer` to lean on DRF as much as possible
- Moved `permissions_group_name_for_role` into a new `website/utils.py` to avoid circular imports
- Removed `group` from `WebsiteCollaboratorSerializer` since it's unused and is a backend detail that the frontend doesn't need to know about

#### How should this be manually tested?
Go to the collaborators UI and verify you can view/add/edit/delete editors and administrators.
